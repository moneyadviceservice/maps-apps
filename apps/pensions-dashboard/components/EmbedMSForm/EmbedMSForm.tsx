import { twMerge } from 'tailwind-merge';

type Props = {
  src: string;
  title: string;
  testId?: string;
  id?: string;
  classNames?: string;
};

export const EmbedMSForm = ({
  src,
  title,
  testId = 'embedded-ms-form',
  id = 'emebedded-ms-form',
  classNames,
}: Props) => {
  return (
    <iframe
      src={src + '&embed=true'}
      title={title}
      data-testid={testId}
      id={id}
      className={twMerge('w-full', classNames)}
    />
  );
};
