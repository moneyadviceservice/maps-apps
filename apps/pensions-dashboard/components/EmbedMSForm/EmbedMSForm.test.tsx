import { render, screen } from '@testing-library/react';

import { EmbedMSForm } from './EmbedMSForm';

import '@testing-library/jest-dom';

describe('EmbedMSForm', () => {
  const defaultProps = {
    src: 'https://example.com/form',
    title: 'Test Form',
  };

  it('renders default iframe', () => {
    const { container } = render(<EmbedMSForm {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders an iframe with a supplied data-testid', () => {
    render(<EmbedMSForm {...defaultProps} testId="test-id" />);
    const iframe = screen.getByTestId('test-id');
    expect(iframe).toBeInTheDocument();
  });

  it('renders an iframe with a different id', () => {
    const { container } = render(
      <EmbedMSForm {...defaultProps} id="test-id" />,
    );
    const iframe = container.querySelector('#test-id');
    expect(iframe).toBeInTheDocument();
  });

  it('applies additional classNames', () => {
    render(<EmbedMSForm {...defaultProps} classNames="extra-class" />);
    const iframe = screen.getByTestId('embedded-ms-form');
    expect(iframe).toHaveClass('w-full extra-class');
  });
});
