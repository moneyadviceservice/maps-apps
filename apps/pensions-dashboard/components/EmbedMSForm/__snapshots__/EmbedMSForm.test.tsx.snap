// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedMSForm renders default iframe 1`] = `
<div>
  <iframe
    class="w-full"
    data-testid="embedded-ms-form"
    id="emebedded-ms-form"
    src="https://example.com/form&embed=true"
    title="Test Form"
  />
</div>
`;
